File: practica8/python/main.py
from abc import ABC, abstractmethod

class Vehiculo(ABC):
    def __init__(self, marca, modelo, year):
        self.marca = marca
        self.modelo = modelo
        self.year = year

    @abstractmethod
    def encender(self):
        pass

    @abstractmethod
    def apagar(self):
        pass

    @abstractmethod
    def describir(self):
        pass

class Auto(Vehiculo):
    def encender(self):
        print(f"El auto {self.marca} {self.modelo} está encendido.")

    def apagar(self):
        print(f"El auto {self.marca} {self.modelo} está apagado.")

    def describir(self):
        print(f"Este es un auto de marca {self.marca}, modelo {self.modelo}, del year {self.year}.")

class Motocicleta(Vehiculo):
    def encender(self):
        print(f"La motocicleta {self.marca} {self.modelo} está encendida.")

    def apagar(self):
        print(f"La motocicleta {self.marca} {self.modelo} está apagada.")

    def describir(self):
        print(f"Esta es una motocicleta de marca {self.marca}, modelo {self.modelo}, del year {self.year}.")

class Camion(Vehiculo):
    def encender(self):
        print(f"El camión {self.marca} {self.modelo} está encendido.")

    def apagar(self):
        print(f"El camión {self.marca} {self.modelo} está apagado.")

    def describir(self):
        print(f"Este es un camión de marca {self.marca}, modelo {self.modelo}, del year {self.year}.")
        
def main():
    vehiculos = []

    vehiculos.append(Auto("Toyota", "Corolla", 2020))
    vehiculos.append(Motocicleta("Yamaha", "MT-07", 2021))
    vehiculos.append(Camion("Ford", "F-150", 2019))

    for vehiculo in vehiculos:
        vehiculo.encender()
        vehiculo.describir()
        vehiculo.apagar()
        print()  # Línea en blanco para separar las descripciones

if __name__ == "__main__":
    main()

File: practica8/java/src/Main.java
import java.util.ArrayList;
import java.util.List;

abstract class Vehiculo {
    protected String marca;
    protected String modelo;
    protected int year;

    public Vehiculo(String marca, String modelo, int year) {
        this.marca = marca;
        this.modelo = modelo;
        this.year = year;
    }

    public abstract void encender();
    public abstract void apagar();
    public abstract void describir();
}

class Auto extends Vehiculo {

    public Auto(String marca, String modelo, int year) {
        super(marca, modelo, year);
    }

    @Override
    public void encender() {
        System.out.println("El auto " + marca + " " + modelo + " está encendido.");
    }

    @Override
    public void apagar() {
        System.out.println("El auto " + marca + " " + modelo + " está apagado.");
    }

    @Override
    public void describir() {
        System.out.println("Este es un auto de marca " + marca + ", modelo " + modelo + ", del year " + year + ".");
    }
}

class Motocicleta extends Vehiculo {

    public Motocicleta(String marca, String modelo, int year) {
        super(marca, modelo, year);
    }

    @Override
    public void encender() {
        System.out.println("La motocicleta " + marca + " " + modelo + " está encendida.");
    }

    @Override
    public void apagar() {
        System.out.println("La motocicleta " + marca + " " + modelo + " está apagada.");
    }

    @Override
    public void describir() {
        System.out.println("Esta es una motocicleta de marca " + marca + ", modelo " + modelo + ", del year " + year + ".");
    }
}

class Camion extends Vehiculo {

    public Camion(String marca, String modelo, int year) {
        super(marca, modelo, year);
    }

    @Override
    public void encender() {
        System.out.println("El camión " + marca + " " + modelo + " está encendido.");
    }

    @Override
    public void apagar() {
        System.out.println("El camión " + marca + " " + modelo + " está apagado.");
    }

    @Override
    public void describir() {
        System.out.println("Este es un camión de marca " + marca + ", modelo " + modelo + ", del year " + year + ".");
    }
}



public class Main {
    public static void main(String[] args) {
        List<Vehiculo> vehiculos = new ArrayList<>();

        vehiculos.add(new Auto("Toyota", "Corolla", 2020));
        vehiculos.add(new Motocicleta("Yamaha", "MT-07", 2021));
        vehiculos.add(new Camion("Ford", "F-150", 2019));

        for (Vehiculo vehiculo : vehiculos) {
            vehiculo.encender();
            vehiculo.describir();
            vehiculo.apagar();
            System.out.println();
        }
    }
}